core/Pos/Core/Timestamp.hs

```diff
@@ -59,6 +61,9 @@ diffTimestamp t1 t2 = getTimestamp t1 - getTimestamp t2
 addMicrosecondsToTimestamp :: Microsecond -> Timestamp -> Timestamp
 addMicrosecondsToTimestamp m t = Timestamp { getTimestamp = (getTimestamp t) + m }
 
+timestampToPosix :: Timestamp -> POSIXTime
+timestampToPosix (Timestamp ts) = fromIntegral ts / 1000000
```

Comment:
Hm, it may work.
But `POSIXTime` is not integer actually. And it's not clear which type to use instead of `Integer` not to loose precision accidentally (in general case)